src/jailer/src/env.rs

```diff
@@ -184,6 +189,41 @@ impl Env {
         self.uid
     }
 
+    fn exec_into_new_pid_ns(&self, chroot_exec_file: PathBuf) -> Result<()> {
+        // Unshare into a new PID namespace.
+        // The current process will not be moved into the newly created namespace, but its first
+        // child will assume the role of init(1) in the new namespace.
+        // The call is safe because we're invoking a C library function with valid parameters.
+        SyscallReturnCode(unsafe { libc::unshare(libc::CLONE_NEWPID) })
+            .into_empty_result()
+            .map_err(Error::UnshareNewPID)?;
+
+        // Duplicate the current process. The child process will belong to the previously created
+        // PID namespace.
+        let pid = unsafe { libc::fork() };
+        match pid {
+            0 => Err(Error::Exec(self.exec_command(chroot_exec_file))),
+            child_pid => Env::wait_child_process(child_pid),
```

Comment:
Why should we keep this process open waiting for the child? Why not just terminate the parent?

Reply:
For testing purposes, mainly. Terminating the parent here would make it hard to test that the child process has been indeed spawned in a PID namespace that is different from its parent's.

Reply:
we are currently spawning the jailer into different PID namespace. we are using `/proc/<pid>/status`'s `﻿NStgid` value in integration testing to find PID namespace is nested and its PID is 0. Hope it will be useful here.

Reply:
Indeed, let's try it this way, or find another way. We can't keep all these processes around doing nothing, they can incur a cost at scale.